frontend/src/components/Counters/Counters.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+import { useEffect, useState, useRef } from "react";
+
+//project resources
+import { getChainsLength } from "../../util/firebase/chain";
+import SingleCounter from "./SingleCounter";
+
+//mui
+import { makeStyles } from "@material-ui/core";
+import theme from "../../util/theme";
+import { number } from "yup/lib/locale";
+import ArrowDownwardIcon from "@mui/icons-material/ArrowDownward";
+
+const Counters = () => {
+  const classes = makeStyles(theme as any)();
+  const containerRef = useRef(null);
+
+  const [chains, setChains] = useState(0);
+
+  const [isVisible, setIsVisible] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      const chainsLengthResponse = await getChainsLength();
+      setChains(chainsLengthResponse);
+    })();
+  }, []);
+
+  //check if div is visible on viewport
+  const callBAck = (entries: any) => {
+    const [entry] = entries;
+    setIsVisible(entry.isIntersecting);
+  };
+
+  const options = {
+    root: null,
+    rootMargin: "50px",
+    threshold: 0.5,
+  };
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(callBAck, options);
+    if (containerRef.current) observer.observe(containerRef.current);
+
+    return () => {
+      if (containerRef.current) observer.unobserve(containerRef.current);
+    };
```

Comment:
useEffect should generally only be used for code that has side-effects (like networking, console, io, changing browser state, etc). Are you sure it's necessary here?